cloudmock/aws/mockautoscaling/group.go

```diff
@@ -18,52 +18,129 @@ package mockautoscaling
-	newGroup := &autoscaling.Group{
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	glog.V(2).Infof("CreateAutoScalingGroup %v", input)
+	createdTime := time.Now().UTC()
+
+	g := &autoscaling.Group{
 		AutoScalingGroupName: input.AutoScalingGroupName,
-		MinSize:              input.MinSize,
-		MaxSize:              input.MaxSize,
-		Instances:            []*autoscaling.Instance{},
+		AvailabilityZones:    input.AvailabilityZones,
+		CreatedTime:          &createdTime,
+		DefaultCooldown:      input.DefaultCooldown,
+		DesiredCapacity:      input.DesiredCapacity,
+		// EnabledMetrics:          input.EnabledMetrics,
+		HealthCheckGracePeriod:  input.HealthCheckGracePeriod,
+		HealthCheckType:         input.HealthCheckType,
+		Instances:               []*autoscaling.Instance{},
+		LaunchConfigurationName: input.LaunchConfigurationName,
+		LoadBalancerNames:       input.LoadBalancerNames,
+		MaxSize:                 input.MaxSize,
+		MinSize:                 input.MinSize,
+		NewInstancesProtectedFromScaleIn: input.NewInstancesProtectedFromScaleIn,
+		PlacementGroup:                   input.PlacementGroup,
+		// Status:                           input.Status,
+		// SuspendedProcesses:               input.SuspendedProcesses,
+		// Tags:                input.Tags,
+		TargetGroupARNs:     input.TargetGroupARNs,
+		TerminationPolicies: input.TerminationPolicies,
+		VPCZoneIdentifier:   input.VPCZoneIdentifier,
 	}
 
-	m.Groups = append(m.Groups, newGroup)
-	return nil, nil
-}
+	for _, tag := range input.Tags {
+		g.Tags = append(g.Tags, &autoscaling.TagDescription{
+			Key:               tag.Key,
+			PropagateAtLaunch: tag.PropagateAtLaunch,
+			ResourceId:        tag.ResourceId,
+			ResourceType:      tag.ResourceType,
+			Value:             tag.Value,
+		})
+	}
 
-func (m *MockAutoscaling) EnableMetricsCollection(input *autoscaling.EnableMetricsCollectionInput) (*autoscaling.EnableMetricsCollectionOutput, error) {
-	return nil, nil
+	if m.Groups == nil {
+		m.Groups = make(map[string]*autoscaling.Group)
+	}
+	m.Groups[*g.AutoScalingGroupName] = g
+
+	return &autoscaling.CreateAutoScalingGroupOutput{}, nil
 }
 
-func (m *MockAutoscaling) DescribeAutoScalingGroups(input *autoscaling.DescribeAutoScalingGroupsInput) (*autoscaling.DescribeAutoScalingGroupsOutput, error) {
-	if len(input.AutoScalingGroupNames) == 0 {
-		return &autoscaling.DescribeAutoScalingGroupsOutput{
-			AutoScalingGroups: m.Groups,
-		}, nil
+func (m *MockAutoscaling) EnableMetricsCollection(request *autoscaling.EnableMetricsCollectionInput) (*autoscaling.EnableMetricsCollectionOutput, error) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	glog.Infof("EnableMetricsCollection: %v", request)
+
+	g := m.Groups[*request.AutoScalingGroupName]
+	if g == nil {
+		return nil, fmt.Errorf("AutoScalingGroup not found")
+	}
+
+	metrics := make(map[string]*autoscaling.EnabledMetric)
```

Comment:
Looks like the same code as in groups? Maybe it is different interfaces but the same business logic.

Reply:
Never mind the rebase messed me ups ignore the comment,